src/Microsoft.AspNetCore.OData/Routing/ODataActionSelector.cs

```diff
@@ -35,21 +41,35 @@ public class ODataActionSelector : IActionSelector
         /// <param name="actionDescriptorCollectionProvider">IActionDescriptorCollectionProvider instance from dependency injection.</param>
         /// <param name="actionConstraintProviders">ActionConstraintCache instance from dependency injection.</param>
         /// <param name="loggerFactory">ILoggerFactory instance from dependency injection.</param>
+        /// <param name="modelBinderFactory">IModelBinderFactory instance from dependency injection</param>
+        /// <param name="modelMetadataProvider">IModelMetadataProvider instance from dependency injection</param>
         public ODataActionSelector(
             IActionDescriptorCollectionProvider actionDescriptorCollectionProvider,
             ActionConstraintCache actionConstraintProviders,
-            ILoggerFactory loggerFactory)
+            ILoggerFactory loggerFactory,
+            IModelBinderFactory modelBinderFactory = null,
+            IModelMetadataProvider modelMetadataProvider = null
+        )
         {
             _innerSelector = new ActionSelector(actionDescriptorCollectionProvider, actionConstraintProviders, loggerFactory);
+            _modelBinderFactory = modelBinderFactory;
+            _modelMetadataProvider = modelMetadataProvider;
         }
 #else
         /// <summary>
         /// Initializes a new instance of the <see cref="ODataActionSelector" /> class.
         /// </summary>
         /// <param name="innerSelector">The inner action selector.</param>
-        public ODataActionSelector(IActionSelector innerSelector)
+        /// <param name="modelBinderFactory">IModelBinderFactory instance from dependency injection</param>
+        /// <param name="modelMetadataProvider">IModelMetadataProvider instance from dependency injection</param>
+        public ODataActionSelector(
+            IActionSelector innerSelector,
+            IModelBinderFactory modelBinderFactory = null,
```

Comment:
> = null [](start = 50, length = 7)

They are from DI, So why makes them as optional parameter?

Reply:
I didn't want to make it a breaking change since it's a public class. But I can remove the `= null` if that's not a problem.

Reply:
I've updated it. The parameters are no longer optional.